Templates/templates/Java/requests_extensions/BaseEntityRequest.java.tt

```diff
@@ -230,10 +235,11 @@ import <#=importNamespace#>.http.HttpMethod;
      * Creates a {0} with a new object
      *
      * @param new{0} the new object to create
-     * @param callback the callback to be called after success or failure
+     * @return a future with the result
      */
-    public void post(@Nonnull final {0} new{0}, @Nonnull final ICallback<? super {0}> callback) {{
-        send(HttpMethod.POST, callback, new{0});
+    @Nonnull
+    public java.util.concurrent.CompletableFuture<{0}> postAsync(@Nonnull final {0} new{0}) {{
```

Comment:
There are POST scenarios that result in a 204 No Content so that content can be null. Overall, I'm hesitant to attribute with NotNull unless the metadata is explict.

Reply:
Same comment as above. Futures are the equivalent of task in dot net. It's not because that task doesn't return anything that you don't have a task